client/QuickStart/API_Util.cs

```diff
@@ -0,0 +1,410 @@
+                // This is mandatory for some scenarios (such as cancelling queries) and will make troubleshooting easier in others.
+                ClientRequestId = $"{scope};{Guid.NewGuid().ToString()}"
+            };
+            // Tip: Though uncommon, you can alter the request default command timeout using the below command, e.g. to set the timeout to 10 minutes, use "10m"
+            if (timeout != null)
+                clientRequestProperties.SetOption(ClientRequestProperties.OptionServerTimeout, timeout);
+
+            return clientRequestProperties;
+        }
+
+        /// <summary>
+        /// Executes a Command using a premade client
+        /// </summary>
+        /// <param name="client">Premade client to run Commands. can be either an adminClient or queryClient</param>
+        /// <param name="configDatabaseName">DB name</param>
+        /// <param name="command">The Command to execute</param>
+        /// <returns>True on success, false otherwise</returns>
+        public static async Task ExecuteAsync(IDisposable client, string configDatabaseName, string command)
+        {
+            try
+            {
+                Newtonsoft.Json.Linq.JObject[] result;
+                if (command.StartsWith(MgmtPrefix)) // All control commands start with a specific prefix (usually '.') - and require a different client and scope to execute with.
+                {
+                    var clientRequestProperties = CreateClientRequestProperties("CS_SampleApp_ControlCommand");
+                    ICslAdminProvider adminClient = (ICslAdminProvider)client;
+                    result = (await adminClient.ExecuteControlCommandAsync(configDatabaseName, command, clientRequestProperties)).ToJObjects().ToArray();
+                }
+                else
+                {
+                    var clientRequestProperties = CreateClientRequestProperties("CS_SampleApp_Query");
+                    ICslQueryProvider queryClient = (ICslQueryProvider)client;
+                    result = (await queryClient.ExecuteQueryAsync(configDatabaseName, command, clientRequestProperties)).ToJObjects().ToArray();
+                }
+
+                // Tip: Actual implementations wouldn't generally print the response from a control command or a query .We print here to demonstrate what a sample of the response looks like.
+                // Moreover, there are some built-in classes for control commands under the Kusto.Data namespace -for example, Kusto.Data.TablesShowCommandResult maps to the result of the ".show tables" commands
+                Console.WriteLine($"Response from executed command '{command}':\n--------------------");
+                var firstRow = result[0];
+                foreach (var item in firstRow.Properties())
+                    Console.WriteLine(item);
+            }
+            catch (Exception ex)
+            {
+                var err = ex.GetType().ToString();
```

Comment:
DO NOT DO THIS - this isn't python, don't compare types by name.
use `if ex is KustoClientException kcs { ...}`

(same for other places)

Reply:
Good. but brackets here and elsewhere in the file